page.tsx

```diff
@@ -19,6 +21,20 @@ export default function createPage(swingsetOptions: SwingsetOptions = {}) {
     const baseRoute = useBaseRoute()
     const [filterValue, setFilterValue] = useState<string | undefined>()
     const searchInputRef = useRef<HTMLInputElement>(null)
+    const [isFullscreen, setIsFullscreen] = useState(false)
+    const handleFullscreenBttnClick = () => {
+      router.replace({
+        query: { ...router.query, isFullscreen: !isFullscreen }
+      })
+    }
+
+    useEffect(() => {
+      if (router.query.isFullscreen && router.query.isFullscreen === 'true') {
+        setIsFullscreen(true)
+      } else {
+        setIsFullscreen(false)
+      }
+    }, [router.query.isFullscreen])
```

Comment:
I think we can derive the isFullscreen state from the router vs needing to use the intermediary useState and useEffect.

```tsx
const isFullscreen = router.query.isFullscreen && router.query.isFullscreen === 'true'
```